src/exportStrategies.js

```diff
@@ -129,14 +130,21 @@ export const CSVStream = async ({
       logger('CSVStream', `${records + chunk.length} of ${totalRecords}`)
 
       let formattedData = format(formatRules)(chunk)
-      // Extract column names from first object
-      if (_.isEmpty(columnKeys))
-        columnKeys = extractKeysFromFirstRow(formattedData)
+
+      // If no includeColumns ware passed get them from the first row
+      // this is not accurate and only works in the case where the first row has all the data for all columns
+      if (_.isEmpty(includeColumns)) {
+        // Extract column names from first object
+        columnHeaders = extractKeysFromFirstRow(formattedData)
+      } else {
+        // Start Case the passed includeColumns
+        columnHeaders = convertColumns(includeColumns)
+      }
       // Convert data to CSV rows
-      let rows = convertData(formattedData, columnKeys)
+      let rows = convertData(formattedData, columnHeaders)
```

Comment:
I can change the parameter name in `convertData` if that is the issue here. The actual code in `convertData` does not care what is passed at all since it just uses it to iterate over:
```
// Convert array of objects to array of arrays
export let convertData = (data, columnKeys) => {
  // Extract data from object
  let transformRow = row => _.map(key => _.get(key, row), columnKeys)
  return _.map(transformRow, data)
}
```

Reply:
Since `convertColumns` maps with `_.startCase`, my expectation is that this would be broken in some cases now that we're calling `convertColumns` earlier because it would, for example, try to lookup `row['Line Item']` instead of `row['LineItem']`. I think this is what @dubiousdavid was referring to when he was talking about headers vs keys - my interpretation is that the header is the label whereas the key is the field name.

I think `formattedData` might already have the mapped column names though which is probably why this change was added, but it seems cleaner to avoid the transform for every record anyway (because we'd be running `_.startCase` on every key on every record when we probably only need to do that transformation once for the headers)

Reply:
`cleaner to avoid the transform for every record` that is a good point. Ok will refactor to not do that.